### PR TITLE
refactor: Simplify how active themes are found

### PR DESCRIPTION
**Pull request by Artur-**

### CODE REVIEW

**Review by mshabarov — CHANGES_REQUESTED** [review]

**Review by mshabarov — COMMENTED** [review]
My previous version was quite crappy comparing to this one :)
Looks much more better - less codes, theme manipulations are delegated to util class.

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/ThemeUtils.java [hunk 1]
```diff
@@ -103,26 +106,23 @@ public static Optional<Theme> getThemeAnnotation(VaadinContext context) {
...
     }
 
-    public static Optional<JsonObject> getThemeJsonForTheme(
-            File themesParentFolder, String themeName) throws IOException {
-        File themeJsonFile = Path.of(themesParentFolder.getAbsolutePath(),
-                Constants.APPLICATION_THEME_ROOT, themeName, "theme.json")
-                .toFile();
+    public static Optional<JsonObject> getThemeJson(File frontendFolder,
+            String themeName) throws IOException {
+        File themeFolder = getThemeFolder(frontendFolder, themeName);
```
[the remark this answers is not in the document]
Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/ThemeUtils.java:111]: Do we have a test for this? I assumed this was caught earlier

[the remark this answers is not in the document]
Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/ThemeUtils.java:111]: or should maybe `getThemeFolder` actually throw an exception?

[the remark this answers is not in the document]
Comment (mshabarov) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/ThemeUtils.java:111]: It is caught by TaskUpdateThemeImport which runs after re-bundle checker. Most like no test for this so far.

[the remark this answers is not in the document]
Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/ThemeUtils.java:111]: Added throwing

[the remark this answers is not in the document]
Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/ThemeUtils.java:111]: But now the TaskRunDevBundleBuildTest needs fixing still

**Review by mshabarov — APPROVED** [review]

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/ThemeUtils.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/ThemeUtils.java:111]: Fixed test

**Review by mshabarov — APPROVED** [review]

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/ThemeUtils.java [hunk 2]
```diff
@@ -137,14 +134,77 @@ public static Optional<String> getParentThemeName(JsonObject themeJson) {
...
+        if (themeInProject.exists()) {
+            return themeInProject;
+        }
+
+        File themeFromJar = new File(packagedThemesFolder, themeName);
+        if (themeFromJar.exists()) {
+            return themeFromJar;
+        }
+
+        throw new IllegalArgumentException("The theme folder for the '"
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/ThemeUtils.java:193]: Theme name is empty in one of the failed tests.
Probably we need to handle this case as well:
```
ERROR com.vaadin.flow.server.frontend.TaskRunDevBundleBuild - Error when checking if a development bundle build is needed
java.lang.IllegalArgumentException: The theme folder for the '' theme was not found. It should be either in ...
```

**Review by mshabarov — CHANGES_REQUESTED** [review]
.

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/ThemeUtils.java (hunk 2 above)
Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/ThemeUtils.java:193]: Yes, I thought I fixed it but I will fix the fix to fix it now

Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/ThemeUtils.java:193]: I think I fixed the fix nwo

**Review by mshabarov — APPROVED** [review]

**Pull request merged**